client/client.go

```diff
@@ -538,3 +542,32 @@ func (c *Client) Collect(ch chan<- prometheus.Metric) {
 var (
 	_ prometheus.Collector = (*Client)(nil)
 )
+
+// argListFromMySqlParams creates an array of strings from the pointer to the parameters for pt-mysql-sumamry
+func argListFromMySqlParams(pParams *agentpb.StartActionRequest_PTMySQLSummaryParams) []string {
+	var args []string
+
+	// Only adds the arguments are valid
+
+	if pParams.Address != "" {
+		args = append(args, "--host="+pParams.Address)
+	}
+
+	if pParams.Port > 0 && pParams.Port <= 65535 {
+		args = append(args, "--port="+strconv.FormatUint(uint64(pParams.Port), 10))
+	}
+
+	if pParams.Socket != "" {
```

Comment:
Can we use socket AND host and port? 
Shouldn't we check and use only one of them?

Reply:
I believe that if the one of them is not used, the string is empty or 0 in case of the port.
If there is another pattern, I can change it but I'd need clear specification to be able to create right conditions. Can you provide me with such information?

Reply:
I would check that if host and port are valid and also we have socket, just use socket and don't use host+port

Reply:
If socket is valid => use just socket
If socket is not valid => check user and password and use them without socket.
Correct?

Reply:
:+1: